tests/test_integration.py

```diff
@@ -116,19 +149,8 @@ def do_GET(self):
         assert MockServerRequestHandler.num_requests == 2
 
     def test_hits_proxy_through_custom_client(self):
```

Comment:
This test doesn't seem to test what it claims

Reply:
```
        stripe.default_http_client = (
            stripe.http_client.new_default_http_client(
                proxy="http://localhost:%s" % self.mock_server_port
            )
        )
 ```
 is creating a new client with a proxy? In any case, I just want to refactor the file so that adding new tests is possible. Reimagining existing tests can stay out of scope.